### PR TITLE
feat(actions): add support for dynamic defaults

### LINKED ISSUE #138: Support default.md fallback
When requesting `/authors/lars.html`, `dispatch` should not just request `/authors/lars.md`, but also `/authors/default.md`, so that we can create a page that says "this author does not exist".

### PR DESCRIPTION
**Pull request by trieloff**
BREAKING CHANGE: Dynamic defaults are a new way of providing placeholders for content that should exist, but isn't fully fleshed out yet. Examples include a tag detail page or author detail page for a blog, when said defaults haven't been provided yet. In the case of requesting `/authors/lars.html`, the corresponding URL is requested, an additional request is made as if the URL was `/authors/default.html`, which will then start the action invocation and trigger loading `/authors/default.md` which can be a placeholder page for authors that for instance fetches all blog posts by this author, but doesn't show a biography. The dyanamic default has higher precedence than a 404 fallback and works for the current directory only. It adds one additional invocation to each request.

fixes #138 but for repositories that already have a `default.md`, this will be served instead of a 404 error page for missing pages in the same directory. The breaking change will not become visible until `helix-publish` is updated to use `@v2` of `helix-dispatch`

### CODE REVIEW

**Comment by trieloff** [comment]
I wouldn't see it as loading a different resource or a non-existing URL. It's loading the dynamic default content that exists at the specified URL, it's just that the dynamic default content is provided by either a markdown document (or some more complex logic in the action, if you'd like).

If you want redirect behavior (which I can see as desirable in a few scenarios), you can implement this in your `pre.js`. As for the default behavior, I'm with @davidnuescheler and think we should just serve the dynamic default content without any indication to the visitor that it somehow is a placeholder or dynamically assembled.

**Comment by tripodsan** [comment]
@davidnuescheler I think a redirect to a `default.html` is better than serving the content of `default.md` under the author's page. this has implications for indexing (internal and external).

also, it can lead to a potential problem when attackers request many false authors.

File: src/fetchers.js [hunk 1]
```diff
@@ -67,34 +79,56 @@ function staticaction(contentOpts) {
  * @property {string} [branch] the optional branch or tag name
  */
 
+function getDefault(url) {
```
Comment (tripodsan) [inline src/fetchers.js]: would be nice to add a jsdoc...

**Comment by tripodsan** [comment]
btw, I changed my mind. I think having the original URL available in the client is beneficial.  as @trieloff mentioned, a redirect could still be implemented in the `default.html` action.

File: src/fetchers.js (hunk 1 above)
Comment (trieloff) [reply src/fetchers.js]: Added in 26187a4

**Pull request merged**